src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -12,19 +11,30 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int defaultBufferSize = 4096;
+        private const int DefaultBufferSize = 4096;
 
         private Stream _content;
+        private CancellationToken _cancellationToken;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, defaultBufferSize)
+            : this(content, DefaultBufferSize)
         {
         }
 
         public StreamContent(Stream content, int bufferSize)
+            : this(content, bufferSize, CancellationToken.None)
+        {
+        }
+
+        internal StreamContent(Stream content, CancellationToken cancellationToken)
```

Comment:
> That would be the best way to pass in a token.

Yup, agreed.  This ctor approach is a temporary workaround for the lack of such APIs.

I suggest we go ahead with this change, and then this small piece of it can be undone when such overloads are available; if nothing else, it'll serve as a good test case to help validate the design.